contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -810,31 +810,36 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
       (bool exist, uint32 roundId) = _doesReportExist(report_);
       // In case the report exists, copy the round id and pay the transmitter
       if (exist) {
+        // In case the round has already been processed by the secondary feed
+        if (s_hotVars.latestSecondaryRoundId >= roundId) {
+          revert StaleReport();
+        }
+
         s_hotVars.latestSecondaryRoundId = roundId;
         emit SecondaryRoundIdUpdated(roundId);
 
         _payTransmitter(s_hotVars, report_.juelsPerFeeCoin, uint32(initialGas), msg.sender);
         return;
       }
-      // In case the report doesn't exist, lock the primary feed
-      s_primaryLocked = true;
-    } else {
```

Comment:
suggestion: I think we could keep this else and use it for the case where the primary feed deals with receiving an existing, but latest report.

```if (epochAndRound == s_hotVars.latestEpochAndRound && s_primaryLocked))
```

I see this if, else block as where the logic is evaluated for all edge cases where an existing report is sent to either feed. Early returns would happen here as well.

The rest of the function would be to handle and record new reports.

This is very subjective so feel free to disagree.

Reply:
Could you develop more this idea? Not sure I'm following how it should be implemented.